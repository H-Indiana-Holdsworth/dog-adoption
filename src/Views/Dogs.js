import React from 'react';
import { useEffect, useState } from 'react';
import DogList from '../Components/DogList';
import { fetchDogs } from '../services/service-dogs';

export default function Dogs() {
  const [dogs, setDogs] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const resp = await fetchDogs();
      setDogs(resp);
    };
    fetchData();
  }, []);

  return (
    <div>
      <DogList dogs={dogs} />
    </div>
  );
}
