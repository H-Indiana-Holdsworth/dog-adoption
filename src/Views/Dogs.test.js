import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Dogs from './Dogs';

test('should render Dogs component', async () => {
  const { container } = render(
    <MemoryRouter>
      {/* <Dogs
        dogs={{
          id: 2,
          name: 'Myra',
          bio: 'Shy.',
          breed: 'Shepard mix',
          image: 'https://placedog.net/500?id=73',
        }}
      /> */}
      <Dogs
        dogs={{
          header: 'Meet Myra',
        }}
      />
    </MemoryRouter>
  );

  await screen.findByText('Meet Myra');

  expect(container).toMatchSnapshot();
});
