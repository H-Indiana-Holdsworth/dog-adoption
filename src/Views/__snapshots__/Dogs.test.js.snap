// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dogs component 1`] = `
<div>
  <div>
    <div
      class="dog-list"
    >
      <div>
        <h2
          class="header-name"
        >
          Meet 
          Myra
        </h2>
        <a
          href="/dogs/2"
        >
          <div>
            <img
              class="dog-img"
              src="https://placedog.net/500?id=73"
            />
            <p
              class="info-link"
            >
              Myra
               is a 
              2
               year old 
              Shepard mix
            </p>
          </div>
        </a>
      </div>
      <div>
        <h2
          class="header-name"
        >
          Meet 
          Pindy
        </h2>
        <a
          href="/dogs/3"
        >
          <div>
            <img
              class="dog-img"
              src="https://placedog.net/500?id=23"
            />
            <p
              class="info-link"
            >
              Pindy
               is a 
              8
               year old 
              Terrier
            </p>
          </div>
        </a>
      </div>
      <div>
        <h2
          class="header-name"
        >
          Meet 
          Jandroo
        </h2>
        <a
          href="/dogs/4"
        >
          <div>
            <img
              class="dog-img"
              src="https://placedog.net/500?id=20"
            />
            <p
              class="info-link"
            >
              Jandroo
               is a 
              14
               year old 
              Nike
            </p>
          </div>
        </a>
      </div>
      <div>
        <h2
          class="header-name"
        >
          Meet 
          Lil Mark
        </h2>
        <a
          href="/dogs/5"
        >
          <div>
            <img
              class="dog-img"
              src="https://placedog.net/500?id=41"
            />
            <p
              class="info-link"
            >
              Lil Mark
               is a 
              6
               year old 
              Pug
            </p>
          </div>
        </a>
      </div>
      <div>
        <h2
          class="header-name"
        >
          Meet 
          Kelm Melm
        </h2>
        <a
          href="/dogs/1"
        >
          <div>
            <img
              class="dog-img"
              src="https://placedog.net/500?id=97"
            />
            <p
              class="info-link"
            >
              Kelm Melm
               is a 
              88
               year old 
              German shepard mix
            </p>
          </div>
        </a>
      </div>
      <div>
        <h2
          class="header-name"
        >
          Meet 
          Mandy
        </h2>
        <a
          href="/dogs/16"
        >
          <div>
            <img
              class="dog-img"
            />
            <p
              class="info-link"
            >
              Mandy
               is a 
              3
               year old 
              Shepard
            </p>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
